includes/wikia/GlobalTitle.php

```diff
@@ -70,6 +70,12 @@ public static function newFromText( $text, $namespace = NS_MAIN, $city_id = null
 		}
 
 		$filteredText = Sanitizer::decodeCharReferences( $text );
+
+		if ( $namespace === NS_SPECIAL ) {
+			$localName = SpecialPageFactory::getLocalNameFor( $text );
```

Comment:
I'd also suggest adding a unit test for localised special page names if possible to catch this.